File: Decode String.cpp
class Solution {
public:
    string decodeString(string s) {
        string res = "";
        stack<char> stack;
        for(char c : s) {
            if(c != ']')
                stack.push(c);
            else {
                string temp = "";
                while(stack.top() != '[') {
                    temp = stack.top() + temp;
                    stack.pop();
                }
                stack.pop();

                string number = "";
                while(!stack.empty() &&  isdigit(stack.top())) {
                    number = stack.top() + number;
                    stack.pop();
                }
                int num = stoi(number);
                for(int i = 0; i < num; i++) {
                    for(char c : temp)
                        stack.push(c);
                }
            }
        }
        while(!stack.empty()) {
            res = stack.top() + res;
            stack.pop();
        }
        return res;
    }
};

File: Decode String.py
class Solution:
    def decodeString(self, s: str) -> str:
        stack = []
        for c in s:
            if c != ']':
                stack.append(c)
            else:
                substr = ""
                while stack[-1] != '[':
                    substr = stack.pop() + substr
                stack.pop()

                K = ""
                while stack and stack[-1].isdigit():
                    K = stack.pop() + K
                stack.append(int(K) * substr)
        return "".join(stack)
             

        
